CONTRIBUTING.md

```diff
@@ -41,27 +37,130 @@ Changes are welcome from all members of the Community.
 * varchar2 lengths are set in characters not bytes 
 
 
-## Testing Environment ##
+## Configuring local environment ##
+
+Your local environment can be of any flavor (Unix/Linux/Windows/Mac). 
+At minimum you need to have Oracle database 11.2 XE accessible for the project and SYS account access to install and develop utPLSQL.
+
+We use four different database accounts (users) for development process.
+* `ut3_latest_release` - holds latest released version of utPLSQL. This schema holds the testing framework used for self-testing of utPLSQL development.
+* `ut3` - holds latest (current) development version of utPLSQL. This is the schema you will be working on.
+* `ut3_tester` - holds unit test packages for development of utPLSQL.
+* `ut3$user#` - used for testing accessibility to schema names with special characters.
+
+
+
+Snippet to get you started with development.
+ 
+_If you're using Windows, you can run below scripts using `GIT bash` - Windows-based Unix-like command line._
+
+```bash
+# clone your fork of utPLSQL
+git clone https://github.com/your account/utPLSQL.git utPLSQL
+
+cd utPLSQL
+
+# add main porject repo as upstream
+git remote add upstream https://github.com/utPLSQL/utPLSQL.git
+
+# fetch all remote repositories
+git fetch --all
+
+# clone utPLSQL master branch from upstream into utPLSQL sub-directory of your project
+git clone --depth=1 --branch=master https://github.com/utPLSQL/utPLSQL.git
+
+# download beta version of utPLSQL-cli
+curl -Lk -o utPLSQL-cli.zip https://bintray.com/viniciusam/utPLSQL-cli/download_file?file_path=utPLSQL-cli-develop-test3.zip
+# unzip utPLSQL-cli and remove the zip file
+unzip utPLSQL-cli.zip && chmod -R u+x utPLSQL-cli && rm utPLSQL-cli.zip 
+
+```
+
+Refreshing your local repo.
+```bash
+# fetch all remote repositories
+git fetch --all
+
+# remove sub-direcotry containing master branch shallow copy
+rm -rf utPLSQL/*
+# clone utPLSQL master branch from upstream into utPLSQL sub-directory of your project
+git clone --depth=1 --branch=master https://github.com/utPLSQL/utPLSQL.git
+
+rm -rf utPLSQL-cli/*
+# download beta version of utPLSQL-cli
+curl -Lk -o utPLSQL-cli.zip https://bintray.com/viniciusam/utPLSQL-cli/download_file?file_path=utPLSQL-cli-develop-test3.zip
+# unzip utPLSQL-cli and remove the zip file
+unzip utPLSQL-cli.zip && chmod -R u+x utPLSQL-cli && rm utPLSQL-cli.zip 
+
+```
+
+Cleanup of utPLSQL installation.
+```sql
+drop user ut3 cascade;
+drop user ut3_latest_release cascade;
+drop user ut3_tester cascade;
+drop user ut3$user# cascade;
+
+begin
+  for i in (
+    select decode(owner,'PUBLIC','drop public synonym "','drop synonym "'||owner||'"."')|| synonym_name ||'"' drop_orphaned_synonym from dba_synonyms a
+     where not exists (select null from dba_objects b where a.table_name=b.object_name and a.table_owner=b.owner )
+  ) loop
+    execute immediate i.drop_orphaned_synonym;
+  end loop;
+end;
+/
```

Comment:
Doesn't work inside 12c PDBs, raises ORA-65040: operation not allowed from within a pluggable database

Reply:
any alternative or any idea how to get this to work?

Reply:
Sorry I didn't come with a suggestion in the first place.
Problem is that it tries to drop SYS-Synonyms, which is not allowed for PDBS.
The following should do it:

```
begin
  for i in (
    select decode(owner,'PUBLIC','drop public synonym "','drop synonym "'||owner||'"."')|| synonym_name ||'"' drop_orphaned_synonym from dba_synonyms a
     where a.table_owner <> 'SYS' and not exists (select null from dba_objects b where a.table_name=b.object_name and a.table_owner=b.owner )
  ) loop
    execute immediate i.drop_orphaned_synonym;
  end loop;
end;
```